socs/agents/orientalmotor_blh/drivers.py

```diff
@@ -0,0 +1,201 @@
+        assert self._connected
+        assert len(data) == 35
+        tmpdt = datetime.now(tz=UTC)
+
+        # Duration from the connection establishment
+        ds_td = tmpdt - self._start_dt
+        ds_int = int((ds_td.total_seconds() * 1e3) % 65536)
+        ds_b = ds_int.to_bytes(2, 'little')
+
+        # Packet index modulo 256
+        pi_b = (self._packet_index % 256).to_bytes(1, 'little')
+
+        # Packet creation
+        pkt_pre = b'\xff' + data + ds_b + pi_b
+
+        # The last byte is a parity
+        parity = calc_parity(pkt_pre)
+        pkt = pkt_pre + int(parity).to_bytes(1, 'little')
+
+        self.__w(pkt)
+        self._packet_index += 1
+
+        return self.__r(readlen)
+
+    def set_op_number(self, num):
+        '''Set operation number
+
+        Parameter
+        ---------
+        num : int
+            Operation number
+        '''
+        assert 0 <= num < 16
+        mark_0 = (((0b110 << 4) + num) << 5).to_bytes(2, 'little')
+        mark_1 = (((num + 4) % 8) << 5) + (5 - int(num / 8))
+        data = bytearray([0x03, 0x00, 0x01, 0x0a, 0x00, 0x86, 0x01]) + mark_0
+        data += bytearray([0x04, 0x00, 0x00, mark_1]) + bytearray([0] * 22)
+
+        return self._wr(data)
+
+    def set_speed(self, speed):
+        '''Set speed
+
+        Parameter
+        ---------
+        speed : int
+            Speed in RPM
+        '''
+        assert 50 <= speed <= 3000
+
+        data = bytearray([0x03, 0x00, 0x01, 0x0c, 0x00, 0x81, 0x00, 0xc4, 0x01])
+        data += speed.to_bytes(2, 'little')
+        data += bytearray([0] * 3)
+
+        # parity byte inside the packet.
+        p_check = calc_parity(data[3:])
+        data += bytearray([p_check] + [0] * 20)
+
+        return self._wr(data)
+
+    def set_accl_time(self, sec, accl=True):
+        '''Set acceleration time
+
+        Parameters
+        ----------
+        sec : float
+            Acceleration time in seconds
+
+        accl : bool, default True
+            True: Acceleration setting
+            False: Decceleration setting
+        '''
+        assert 0.5 <= sec <= 15.0
+
+        data = bytearray([0x03, 0x00, 0x01, 0x0c, 0x00, 0x81, 0x00, 0xc5 if accl else 0xc6, 0x01])
+        data += int(sec * 10).to_bytes(1, 'little')
+        data += bytearray([0] * 4)
+
+        # parity byte inside the packet
+        p_check = calc_parity(data[3:])
+        data += bytearray([p_check] + [0] * 20)
+        return self._wr(data)
+
+    def get_status(self):
+        '''Get status.
+
+        Returns
+        -------
+        speed : int
+            Speed in RPM
+        '''
+
+        pkt_st = self._wr(STATUS, readlen=80)
+
+        return int.from_bytes(pkt_st[7:11], 'little', signed=True)
+
+    def start(self, forward=True):
+        '''Start rotation.
+
+        Parameter
+        ---------
+        forward : bool, default True
+            Move forward if True
+        '''
+        self._wr(MOVE_FWD if forward else MOVE_BKW)
```

Comment:
If this returns something, you should return it at the end of this function too. This could then be handled in the agent to see whether the command worked as intended.